x-pack/filebeat/module/cisco/ftd/test/security-connection.log-expected.json

```diff
@@ -0,0 +1,707 @@
+        "network.application": "dns client",
+        "network.iana_number": 17,
+        "network.protocol": "dns",
+        "network.transport": "udp",
+        "service.type": "cisco",
+        "source.bytes": 106,
+        "source.ip": "10.0.1.20",
+        "source.packets": 1,
+        "source.port": 50074,
+        "tags": [
+            "cisco-ftd"
+        ],
+        "user.id": "No Authentication Required",
+        "user.name": "No Authentication Required"
+    },
+    {
+        "@timestamp": "2019-08-15T16:07:00.000Z",
+        "cisco.ftd.destination_interface": "outside",
+        "cisco.ftd.list_id": [
+            "default",
+            "Rule-1"
+        ],
+        "cisco.ftd.message_id": "430003",
+        "cisco.ftd.security.ACPolicy": "default",
+        "cisco.ftd.security.AccessControlRuleAction": "Allow",
+        "cisco.ftd.security.AccessControlRuleName": "Rule-1",
+        "cisco.ftd.security.ApplicationProtocol": "DNS",
+        "cisco.ftd.security.Client": "DNS client",
+        "cisco.ftd.security.ConnectionDuration": "0",
+        "cisco.ftd.security.DNSQuery": "siem-inside",
+        "cisco.ftd.security.DNSRecordType": "a host address",
+        "cisco.ftd.security.DNSResponseType": "Non-Existent Domain",
+        "cisco.ftd.security.DNS_TTL": "86395",
+        "cisco.ftd.security.DstIP": "8.8.8.8",
+        "cisco.ftd.security.DstPort": "53",
+        "cisco.ftd.security.EgressInterface": "outside",
+        "cisco.ftd.security.EgressZone": "output-zone",
+        "cisco.ftd.security.IngressInterface": "inside",
+        "cisco.ftd.security.IngressZone": "input-zone",
+        "cisco.ftd.security.InitiatorBytes": "164",
+        "cisco.ftd.security.InitiatorPackets": "2",
+        "cisco.ftd.security.NAPPolicy": "Balanced Security and Connectivity",
+        "cisco.ftd.security.Prefilter Policy": "Default Prefilter Policy",
+        "cisco.ftd.security.Protocol": "udp",
+        "cisco.ftd.security.ResponderBytes": "314",
+        "cisco.ftd.security.ResponderPackets": "2",
+        "cisco.ftd.security.SrcIP": "10.0.1.20",
+        "cisco.ftd.security.SrcPort": "49264",
+        "cisco.ftd.security.User": "No Authentication Required",
+        "cisco.ftd.source_interface": "inside",
+        "destination.as.number": 15169,
+        "destination.as.organization.name": "Google LLC",
+        "destination.bytes": 314,
+        "destination.geo.continent_name": "North America",
+        "destination.geo.country_iso_code": "US",
+        "destination.geo.location.lat": 37.751,
+        "destination.geo.location.lon": -97.822,
+        "destination.ip": "8.8.8.8",
+        "destination.packets": 2,
+        "destination.port": 53,
+        "dns.answers.ttl": "86395",
+        "dns.answers.type": "Non-Existent Domain",
```

Comment:
`dns.answers` should be an array of objects.

But in this specific case, I would argue perhaps you shouldn't even attempt to map answers, as the DNS query failed. So I would suggest setting the error in `dns.response_code`. Non-existent domain is NXDOMAIN.